### PR TITLE
[BIA-109] - Improved .gooddata

### PR DESCRIPTION
**Pull request by korczis**
## What .gooddata file can contain now?
- username 
- password
- authorization token
- environment - *creadentials[:environment] || ENV['GD_ENVIRONMENT'] || GoodData::Project::DEFAULT_ENVIRONMENT*
- server - *creadentials[:server] || ENV['GD_SERVER'] || GoodData::Rest::Connection::DEFAULT_URL*

## Example
```
joe.doe@kx-mac:~/dev/gooddata-ruby$ ./bin/gooddata auth store
Enter your GoodData credentials.
Email
joe.doe@example.com
Password
***************
Authorization Token
YOUR_TOKEN
Environment  |PRODUCTION|
DEVELOPMENT
Server  |https://secure.gooddata.com|
https://staging.getgooddata.com

joe.doe@kx-mac:~/dev/gooddata-ruby$ cat ~/.gooddata
{
  "username": "joe.doe@example.com",
  "password": "sup3rS3cr3tp4zz",
  "auth_token": "YOUR_TOKEN",
  "environment": "DEVELOPMENT",
  "server": "https://staging.getgooddata.com"
}
```

### CODE REVIEW

**Comment by yenkins-admin** [comment]
Merged build triggered.

**Comment by yenkins-admin** [comment]
Merged build started.

**Comment by yenkins-admin** [comment]
Merged build finished.

**Comment by yenkins-admin** [comment]
Test FAILed.
Refer to this link for build results: https://ci.intgdc.com/job/BIA-pull-request-builder/94/

**Comment by yenkins-admin** [comment]
Merged build triggered.

**Comment by yenkins-admin** [comment]
Merged build started.

**Comment by yenkins-admin** [comment]
Merged build finished.

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/BIA-pull-request-builder/95/

File: lib/gooddata/helpers/auth_helpers.rb [hunk 1]
```diff
@@ -31,13 +51,13 @@ def read_credentials(credentials_file_path = credentials_file)
         # @return [String] auth token from .gooddata, environment variable or nil
         def read_token(credentials_file_path = credentials_file)
           goodfile = read_credentials(credentials_file_path)
-          goodfile[:auth_token] || goodfile[:token] || ENV['GD_PROJECT_TOKEN']
+          [goodfile[:auth_token],  goodfile[:token], ENV['GD_PROJECT_TOKEN']].find { |x| !x.nil? && !x.empty? }
```
Comment (fluke777) [inline lib/gooddata/helpers/auth_helpers.rb]: One think I did not notice before. Do not we want precedence of ENV over .gooddata?

**Comment by yenkins-admin** [comment]
Merged build triggered.

**Comment by yenkins-admin** [comment]
Merged build started.

**Comment by yenkins-admin** [comment]
Merged build finished.

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/BIA-pull-request-builder/99/

**Pull request merged**